src/yunohost/migrations.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+
+""" License
+
+    Copyright (C) 2016 YunoHost
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program; if not, see http://www.gnu.org/licenses
+
+"""
+
+""" yunohost_domain.py
+
+    Migration framework for our data modifications.
+"""
+
+import os
+import re
+import json
+import errno
+from importlib import import_module
+from moulinette.utils.log import getActionLogger
+from moulinette.core import MoulinetteError
+
+
+MIGRATIONS_STATE_PATH = "/etc/yunohost/migrations_state.json"
+
+logger = getActionLogger('yunohost.migrations')
+
+
+def migrations_list():
+    """
+    List existing migrations
+    """
+
+    migrations = {"migrations": []}
+
+    for migration in _get_migrations_list():
+        migrations["migrations"].append({
+            "number": int(migration.split("_", 1)[0]),
+            "name": migration.split("_", 1)[1],
+            "file_name": migration,
+        })
+
+    return migrations
+
+
+# TODO need a "fake" option and also a migration number as possible argument (same than in django)
+def migrations_migrate(target=None, fake=False):
+    """
+    Perform migrations
+    """
+
+    # state is a datastructure of that represent the latest runed migration
+    # it has this form:
+    # {
+    #     "last_runed_migration": {
+    #             "number": "00xx",
+    #             "name": "some name",
+    #         }
+    # }
+    state = migrations_state()
+
+    last_runed_migration_number = state["last_runed_migration"]["number"] if state["last_runed_migration"] else 0
+
+    migrations = []
+
+    # loading all migrations
+    for migration in migrations_list()["migrations"]:
+        logger.debug("Loading migration {number} {name}...".format(
+            number=migration["number"],
+            name=migration["name"],
+        ))
+
+        try:
+            module = import_module("yunohost.data_migrations.{file_name}".format(**migration))
+        except Exception:
+            import traceback
+            traceback.print_exc()
```

Comment:
yes, it prints the whole traceback, otherwise only having the exception name is the equivalent of a "fuck you" for debugging.